pennylane/pulse/rydberg.py

```diff
@@ -114,6 +114,147 @@ def circuit():
+
+    .. seealso::
+
+        :func:`~.rydberg_interaction`, :class:`~.ParametrizedHamiltonian`, :class:`~.ParametrizedEvolution`
+        and :func:`~.evolve`
+
+    **Example**
+
+    We create a Hamiltonian describing a laser acting on 4 wires (Rydberg atoms) with a fixed detuning and
+    phase, and a parametrized, time-dependent amplitude. The Hamiltonian includes an interaction term for
+    inter-atom interactions due to van der Waals forces, as well as the driving term for the laser driving
+    the atoms:
+
+    .. code-block:: python
+
+        atom_coordinates = [[0, 0], [0, 4], [4, 0], [4, 4]]
+        wires = [0, 1, 2, 3]
+        H_i = qml.pulse.rydberg_interaction(atom_coordinates, wires)
+
+        amplitude = lambda p, t: p * jnp.sin(jnp.pi * t)
+        detuning = 3 * jnp.pi / 4
+        phase = jnp.pi / 2
+        H_d = qml.pulse.rydberg_drive(amplitude, detuning, phase, wires)
+
+    >>> H_i
+    ParametrizedHamiltonian: terms=6
+    >>> H_d
+    ParametrizedHamiltonian: terms=3
+
+    The first two terms of the drive Hamiltonian ``H_d`` correspond to the first sum (the sine and cosine terms),
+    describing drive between the ground and excited states. The third term corresponds to the shift term
+    due to detuning from resonance. This drive term corresponds to a global drive that acts on all 4 wires of
+    the device.
+
+    The full Hamiltonian can be evaluated:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit.jax", wires=wires)
+        @qml.qnode(dev, interface="jax")
+        def circuit(params):
+            qml.evolve(H_i + H_d)(params, t=[0, 10])
+            return qml.expval(qml.PauliZ(0))
+
+    >>> params = [2.4]
+    >>> circuit(params)
+    Array(0.94301294, dtype=float64)
+    >>> jax.grad(circuit)(params)
+    [Array(0.59484969, dtype=float64)]
+
+    We can also create a Hamiltonian with multiple local drives. The following circuit corresponds to the
+    evolution where an additional local drive acting on wires ``[0, 1]`` is added to the Hamiltonian:
+
+    .. code-block:: python
+
+        amplitude_local = lambda p, t: p[0] * jnp.sin(2 * jnp.pi * t) + p[1]
+        detuning_local = lambda p, t: p * jnp.exp(-0.25 * t)
+        phase_local = jnp.pi / 4
+        H_local = qml.pulse.rydberg_drive(amplitude_local, detuning_local, phase_local, [0, 1])
+
+        H = H_i + H_d + H_local
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def circuit_local(params):
+            qml.evolve(H)(params, t=[0, 10])
+            return qml.expval(qml.PauliZ(0))
+
+    >>> params = [2.4, [1.3, -2.0], -1.5]
+    >>> circuit_local(params)
+    Array(0.80238028, dtype=float64)
+    >>> jax.grad(circuit_local)(params)
+    [Array(-0.73273312, dtype=float64),
+     [Array(-0.45635171, dtype=float64, weak_type=True),
+      Array(0.76572817, dtype=float64, weak_type=True)],
+     Array(0.01027312, dtype=float64)]
+    """
+    if isinstance(wires, int):
```

Comment:
Haven't thought about this before, but are we supposed to care about custom wire labels? In theory someone could do `rydberg_drive(amp, phase, detuning, wires='a')`, right? And then I guess we still want it cast to a list?

Reply:
That's a good point. This should also be updated in @Qottmann 's PR.